test/e2e/runner/benchmark.go

```diff
@@ -46,18 +49,34 @@ func Benchmark(testnet *e2e.Testnet, benchmarkLength int64) error {
 	// slice into time intervals and collate data
 	timeIntervals := splitIntoBlockIntervals(blocks)
 	testnetStats := extractTestnetStats(timeIntervals)
+
+	// populate data
+	testnetStats.populateTxns(blocks)
+	testnetStats.totalTime = dur
+	testnetStats.height = benchmarkLength
 	testnetStats.startHeight = blocks[0].Header.Height
 	testnetStats.endHeight = blocks[len(blocks)-1].Header.Height
 
 	// print and return
 	logger.Info(testnetStats.String())
+	logger.Info(testnetStats.getReportJSON(testnet))
 	return nil
 }
 
+func (t *testnetStats) populateTxns(blocks []*types.BlockMeta) {
+	t.numtxns = 0
+	for _, b := range blocks {
+		t.numtxns += int64(b.NumTxs)
+	}
+}
+
 type testnetStats struct {
 	startHeight int64
 	endHeight   int64
 
+	height    int64
```

Comment:
height is supposed to be the amount of blocks that the benchmark runs over right? Shouldn't subtracting startHeight from endHeight be sufficient?

Reply:
Yeah, I thought so too, but it looks like we pass a "benchmarkLength" to the function was is more like "how long we expect the test to run for" and the start - end is "how many blocks are produced" which can be different, or could provide some useful information about throughput.

Reply:
I think this field name is too ambiguous :-/

Reply:
renamed, to "length" and "benchmarkLength"

Reply:
Que? It's still named height.

Reply:
sorry, branch naming kerfuffle, fixed for real now.